src/Simulation/Common/IQuantumProcessor.cs

```diff
@@ -391,6 +391,41 @@ public interface IQuantumProcessor
         /// <param name="qubits">Register to apply the exponent to.</param>
         void ControlledExpFrac(IQArray<Qubit> controls, IQArray<Pauli> paulis, long numerator, long power, IQArray<Qubit> qubits);
 
+        /// <summary>
+        /// Used as extensibility mechanism to add custom gates to IQuantumProcessor. 
```

Comment:
The term "gate" can be ambiguous, as it is commonly used to refer to quantum operations in general, specifically intrinsic quantum operations, and to unitary operators that represent the action of quantum operations. Given that ambiguity, I would strongly suggest a name that communicates the intent and purpose of this extensibility mechanism; e.g.: `ApplyCustomIntrinsicOperation`.

Reply:
Makes sense. I was thinking to make it more limited, but switching gate to operation is also reasonable. I was thinking about it myself, so if you don't like "Gate" I can certainly understand.

---
In reply to: [410524692](https://github.com/microsoft/qsharp-runtime/pull/172#discussion_r410524692) [](ancestors = 410524692)